### PR TITLE
[frontend] Add button for selecting relationships in knowledge graph

### LINKED ISSUE #8378: Remove group of relation from container by selecting them from a graph
## Use case

Be able to select group of relation (through lasso/any selection method available) relations in graphs in order to remove relations from a container. 
The alternative soltuion is to be able to have a tab displaying the relation contained in the container to removed them from the container.

### PR DESCRIPTION
**Pull request by stefan1anuby**
### Proposed changes

Added a button to select/deselect relationships (children/parent/both) from or to selected nodes in a knowledge graph context.

https://github.com/user-attachments/assets/b7f79311-5571-4a13-8ace-c1db4c2febcd

I'm not entirely sure about the icon choices and would appreciate feedback on UX/UI.

### Related issues

part of https://github.com/OpenCTI-Platform/opencti/issues/8378

### CODE REVIEW

File: opencti-platform/opencti-front/src/private/components/analyses/reports/ReportKnowledgeGraph.jsx [hunk 1]
```diff
@@ -904,6 +904,36 @@ class ReportKnowledgeGraphComponent extends Component {
     this.setState({ numberOfSelectedNodes: this.selectedNodes.size });
   }
 
+  handleSelectRelationshipsByAdjacentNode(type) {
+    const selectedNodes = Array.from(this.selectedNodes);
+    const selectedNodesIds = R.map((n) => n.id, selectedNodes);
```
Comment (Archidoit) [inline opencti-platform/opencti-front/src/private/components/analyses/reports/ReportKnowledgeGraph.jsx]:
```suggestion
    const selectedNodesIds = selectedNodes.map((n) => n.id);
```
(we avoid using Ramda when possible)

File: opencti-platform/opencti-front/src/private/components/analyses/reports/ReportKnowledgeGraph.jsx [hunk 2]
```diff
@@ -904,6 +904,36 @@ class ReportKnowledgeGraphComponent extends Component {
...
+    this.selectedLinks = new Set(
+      Array.from(this.selectedLinks).filter((link) => !R.includes(link.source_id, selectedNodesIds) && !R.includes(link.target_id, selectedNodesIds)),
```
Comment (Archidoit) [inline opencti-platform/opencti-front/src/private/components/analyses/reports/ReportKnowledgeGraph.jsx]:
```suggestion
      Array.from(this.selectedLinks).filter((link) => !selectedNodesIds.includes(link.source_id) && !selectedNodesIds.includes(link.target_id)),
```

Comment (Archidoit) [reply opencti-platform/opencti-front/src/private/components/analyses/reports/ReportKnowledgeGraph.jsx]: same here

File: opencti-platform/opencti-front/src/private/components/analyses/reports/ReportKnowledgeGraph.jsx [hunk 3]
```diff
@@ -904,6 +904,36 @@ class ReportKnowledgeGraphComponent extends Component {
...
+    );
+
+    const relationshipsToSelect = R.filter(
```
Comment (Archidoit) [inline opencti-platform/opencti-front/src/private/components/analyses/reports/ReportKnowledgeGraph.jsx]: same here

**Comment by Archidoit** [comment]
Thanks for your PR ! 👍 
It's working well, I just added some comments for little code improvements.

**Comment by Archidoit** [comment]
Can you also add the translations for the tooltips you added ? ie:
- Select Child Relationships of Selected Nodes (From)
- Select Child Relationships of Selected Nodes (To)
- Deselect Relationships of Selected Nodes
- Select Relationships of Selected Nodes

**Comment by Archidoit** [comment]
Can you also place the new 'select relationships' button just right after the 'select all nodes' button?
In this place : 
image

It's because the right part of the toolbar is for actions button (like editing or deleting nodes) whereas the left part is for selection actions.

**Comment by stefan1anuby** [comment]
> Can you also add the translations for the tooltips you added ? ie:
> 
> * Select Child Relationships of Selected Nodes (From)
> * Select Child Relationships of Selected Nodes (To)
> * Deselect Relationships of Selected Nodes
> * Select Relationships of Selected Nodes

I used Google Translate; I hope this is okay. Also, changed the button position as requested.

**Review by Archidoit — APPROVED** [review]

**Comment by Archidoit** [comment]
Thanks for the changes @stefan1anuby  ! Your 2 last commits are not signed so I can't merge the PR... Can you sign them? Thanks

**Comment by stefan1anuby** [comment]
> Thanks for the changes @stefan1anuby ! Your 2 last commits are not signed so I can't merge the PR... Can you sign them? Thanks

Done! Thank you for the guidance!

**Pull request merged**